zokrates_stdlib/stdlib/utils/casts/field_to_u64.zok

```diff
@@ -1,5 +1,7 @@
-from "EMBED" import unpack, u64_from_bits
+from "field" import FIELD_SIZE_IN_BITS
+import "utils/pack/bool/unpack"
+import "utils/casts/u64_from_bits"
 
-def main(field i) -> u64:
-    bool[64] bits = unpack(i)
-    return u64_from_bits(bits)
\ No newline at end of file
+def main(field input) -> u64:
+    bool[FIELD_SIZE_IN_BITS] bits = unpack(input)
```

Comment:
@m1cm1c this is what we ended up doing: decompose the field to N bits with N the size of the field (254 for bn128) in a safe way (intuition there is to force 0 to be decomposed as [false; 254] and not as the decomposition of P as 254 bits). Then truncate this decomposition of N bits to 64/32/16/8 bits. Please share if you have any thoughts!

Reply:
This sounds like a good approach.